### PR TITLE
Add CRIU security provider

### PR DESCRIPTION
**Pull request by ashbm5**
Signed-off-by: Alon Shalev Housfater <alonsh@ca.ibm.com>

update of https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/494

### CODE REVIEW

**Comment by ashbm5** [comment]
fyi @tajila @JasonFengJ9 @keithc-ca @pshipton

**Comment by JasonFengJ9** [comment]
Jenkins test sanity plinux jdknext

**Review by tajila — APPROVED** [review]

File: src/java.base/share/classes/java/security/Security.java [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/*[INCLUDE-IF JAVA_SPEC_VERSION >= 8]*/
```
Comment (keithc-ca) [inline src/java.base/share/classes/java/security/Security.java:1]: This line is not necessary: please remove it.

File: closed/GensrcJ9JCL.gmk [hunk 2]
```diff
@@ -49,10 +49,14 @@ endif # OPENJ9_ENABLE_OPENJDK_METHODHANDLES
...
 OPENJDK_SOURCE_PROCESS_ENVIRONMENT := $(TOPDIR)/src/java.base/unix/classes/java/lang/ProcessEnvironment.java
 OPENJDK_STAGED_PROCESS_ENVIRONMENT := $(patsubst $(TOPDIR)/%,$(SUPPORT_OUTPUTDIR)/overlay/%,$(OPENJDK_SOURCE_PROCESS_ENVIRONMENT))
-
 $(OPENJDK_STAGED_PROCESS_ENVIRONMENT) : $(OPENJDK_SOURCE_PROCESS_ENVIRONMENT)
 	$(call install-file)
 
+OPENJDK_SOURCE_SECURITY := $(TOPDIR)/src/java.base/share/classes/java/security/Security.java
+OPENJDK_STAGED_SECURITY := $(patsubst $(TOPDIR)/%,$(SUPPORT_OUTPUTDIR)/overlay/%,$(OPENJDK_SOURCE_SECURITY))
+$(OPENJDK_STAGED_SECURITY) : $(OPENJDK_SOURCE_SECURITY)
+	$(call install-file)
```
Comment (keithc-ca) [inline closed/GensrcJ9JCL.gmk:58]: The list of 'overlay' files is likely to grow and should be handled in a more extensible way:
```
# OpenJ9 CRIU only supports Linux, so we only need to consider the share and unix sub-directories.

$(eval $(call SetupCopyFiles,COPY_OVERLAY_FILES, \
	SRC := $(TOPDIR), \
	DEST := $(SUPPORT_OUTPUTDIR)/overlay, \
	FILES := \
		src/java.base/share/classes/java/security/Security.java \
		src/java.base/unix/classes/java/lang/ProcessEnvironment.java \
	))
```

File: closed/GensrcJ9JCL.gmk [hunk 3]
```diff
@@ -76,7 +80,8 @@ IncludeIfUnsure := -includeIfUnsure -noWarnIncludeIf
 
 $(J9JCL_SOURCES_DONEFILE) : \
 		$(foreach dir, $(JppSourceDirs), $(call RecursiveWildcard,$(dir),*.java)) \
-		$(OPENJDK_STAGED_PROCESS_ENVIRONMENT)
+		$(OPENJDK_STAGED_PROCESS_ENVIRONMENT) \
+		$(OPENJDK_STAGED_SECURITY)
```
Comment (keithc-ca) [inline closed/GensrcJ9JCL.gmk:84]: With the suggested change above, this is simplified to `$(COPY_OVERLAY_FILES)`.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/share/classes/java/security/Security.java (hunk 1 above)
Comment (JasonFengJ9) [reply src/java.base/share/classes/java/security/Security.java:1]: Created https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/511
W/ this change, the `INCLUDE-IF` can be removed.

Comment (ashbm5) [reply src/java.base/share/classes/java/security/Security.java]: done

File: closed/GensrcJ9JCL.gmk (hunk 3 above)
Comment (ashbm5) [reply closed/GensrcJ9JCL.gmk]: done

File: closed/GensrcJ9JCL.gmk (hunk 2 above)
Comment (ashbm5) [reply closed/GensrcJ9JCL.gmk]: done

File: closed/GensrcJ9JCL.gmk [hunk 4]
```diff
@@ -46,12 +46,15 @@ ifeq (true,$(OPENJ9_ENABLE_OPENJDK_METHODHANDLES))
   JPP_TAGS += OPENJDK_METHODHANDLES
 endif # OPENJ9_ENABLE_OPENJDK_METHODHANDLES
 
-# OpenJ9 CRIU only supports Linux, so we only need to consider the unix sub-directory.
-OPENJDK_SOURCE_PROCESS_ENVIRONMENT := $(TOPDIR)/src/java.base/unix/classes/java/lang/ProcessEnvironment.java
-OPENJDK_STAGED_PROCESS_ENVIRONMENT := $(patsubst $(TOPDIR)/%,$(SUPPORT_OUTPUTDIR)/overlay/%,$(OPENJDK_SOURCE_PROCESS_ENVIRONMENT))
```
Comment (keithc-ca) [inline closed/GensrcJ9JCL.gmk:49]: Please remove this blank line (we don't need two or more together).

Comment (ashbm5) [reply closed/GensrcJ9JCL.gmk]: done

**Comment by keithc-ca** [comment]
jenkins compile xlinuxcriu jdk11

**Comment by keithc-ca** [comment]
The recent changes should not invalidate the test results in https://openj9-jenkins.osuosl.org/job/PullRequest-OpenJDK11/54.

**Pull request merged**